caps/security_legacy.go

```diff
@@ -0,0 +1,73 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package caps
+
+import (
+	"fmt"
+
+	"github.com/ubuntu-core/snappy/snappy"
+)
+
+// legacySecurtySystem aids in the ongoing work to transition capability system
+// from hwaccess API to a more direct approach. It allows particular capability
+// types to define a common interface that doesn't expose hwaccess API anymore.
+type legacySecuritySystem struct {
+	AttrName string
+}
+
+func (sec *legacySecuritySystem) GrantPermissions(snapName string, cap *Capability) error {
+	const errPrefix = "cannot grant permissions"
+	// Find the snap
+	snap := snappy.ActiveSnapByName(snapName)
+	if snap == nil {
+		return fmt.Errorf("%s, no such package", errPrefix)
+	}
+	// Fetch the attribute where the path is stored
+	path, err := cap.GetAttr(sec.AttrName)
+	if err != nil {
+		return fmt.Errorf("%s, cannot get required attribute: %q", errPrefix, err)
```

Comment:
Here the mismatch becomes obvious. How can the _legacy security system_ know that an arbitrary capability which it has no knowledge about contains a given attribute?